src/main/java/com/textmessenger/service/EmailServiceImpl.java

```diff
@@ -32,11 +40,23 @@ public void sendEmailFromMethods(String mail, String subject, String setText, St
   }
 
   @Override
+  @Async(AsyncConfiguration.TASK_EXECUTOR_SERVICE)
   public void sendEmailFromMethods(String mail, String subject, String text) {
+    CompletableFuture<SimpleMailMessage> email = constructMail(mail, subject, text);
+    try {
+      javaMailSender.send(email.get());
+    } catch (InterruptedException | MailException | ExecutionException e ) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Either re-interrupt this method or rethrow the "InterruptedException". ([squid:S2142](https://sonarcloud.io/organizations/final_project_text_messenger/rules#rule_key=squid%3AS2142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=Text_Messenger_Project&pullRequest=244&issues=AWYNWVa-M63is4RDz4Bg&open=AWYNWVa-M63is4RDz4Bg)

[ISSUE_KEY]: AWYNWVa-M63is4RDz4Bg